internal/io/redis/pubsub/redisPub_test.go

```diff
@@ -0,0 +1,190 @@
+		`{"humidity":82,"status":"wet","temperature":25}`,
+		`{"humidity":60,"status":"hot","temperature":33}`,
+	}
+
+	var actual []string
+	ticker := time.After(10 * time.Second)
+	for i := 0; i < len(exp); i++ {
+		select {
+		case <-ticker:
+			t.Errorf("timeout")
+			return
+		case d := <-ch:
+			actual = append(actual, string(d))
+		}
+	}
+
+	if !reflect.DeepEqual(actual, exp) {
+		t.Errorf("result mismatch:\n\nexp=%#v\n\ngot=%#v\n\n", exp, actual)
+	}
+	time.Sleep(100 * time.Millisecond)
+}
+
+func TestSinkConnExp(t *testing.T) {
+	m, err := miniredis.Run()
+	if err != nil {
+		t.Error(err)
+	}
+	mport := m.Port()
+	s := RedisPub()
+	s.Configure(map[string]interface{}{
+		"address":     m.Addr(),
+		"db":          0,
+		"password":    "",
+		"channel":     DefaultChannel,
+		"compression": "",
+	})
+	data := []interface{}{
+		map[string]interface{}{
+			"temperature": 22,
+			"humidity":    50,
+			"status":      "green",
+		},
+	}
+	ctx := mockContext.NewMockContext("ruleSink", "op1")
+	err = s.Open(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	time.Sleep(time.Second)
+	m.Close()
+	expErrStr := fmt.Sprintf("io error: Error occurred while publishing the Redis message to 127.0.0.1:%s", mport)
+	for _, e := range data {
+		err := s.Collect(ctx, e)
+		if err == nil {
+			t.Errorf("should have error")
+			return
+		} else if err.Error() != expErrStr {
+			t.Errorf("error mismatch:\n\nexp=%s\n\ngot=%s\n\n", expErrStr, err.Error())
+		}
+	}
+	fmt.Println("closing sink")
+	err = s.Close(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSinkConfigure(t *testing.T) {
+	s := RedisPub()
+	prop := map[string]interface{}{
+		"address": "",
+		"db":      "",
+		"channel": DefaultChannel,
+	}
+	expErrStr := fmt.Sprintf("read properties %v fail with error: %v", prop, "1 error(s) decoding:\n\n* 'db' expected type 'int', got unconvertible type 'string', value: ''")
+	err := s.Configure(prop)
+	if err == nil {
+		t.Errorf("should have error")
+		return
+	} else if err.Error() != expErrStr {
+		t.Errorf("error mismatch:\n\nexp=%v\n\ngot=%v\n\n", expErrStr, err.Error())
+	}
+}
+
+func TestSinkDecompressorError(t *testing.T) {
+	s := RedisPub()
+	prop := map[string]interface{}{
+		"address":     "",
+		"db":          0,
+		"channel":     DefaultChannel,
+		"compression": "test",
+	}
+	expErrStr := fmt.Sprintf("invalid compression method test")
+	err := s.Configure(prop)
+	if err == nil {
+		t.Errorf("should have error")
+		return
+	} else if err.Error() != expErrStr {
+		t.Errorf("error mismatch:\n\nexp=%v\n\ngot=%v\n\n", expErrStr, err.Error())
+	}
+}
+
+func TestSinkPingRedisError(t *testing.T) {
+	s := RedisPub()
+	prop := map[string]interface{}{
+		"address": "",
+		"db":      0,
+		"channel": DefaultChannel,
+	}
+	// expErrStr := fmt.Sprintf("Ping Redis failed with error: %v\", err")
+	expErrStr := fmt.Sprintf("Ping Redis failed with error: dial tcp 127.0.0.1:6379: connectex: No connection could be made because the target machine actively refused it.")
```

Comment:
Looks like the error message is different in local and Github env

Reply:
Thank you for bringing this to my attention. It appears that there was a difference in the error message between the local environment and GitHub. I have investigated this issue further, and have already made the necessary changes to resolve it.